### ISSUE TITLE
MIX_TARGET is required to be set when building

### ISSUE THREAD

**Issue by fhunleth** [nerves-project/nerves_bootstrap#24]
Historically, `MIX_TARGET` was just used in the generated `mix.exs` files. There really wasn't any magic and you could just remove it's use if you were only making builds using one system. Now it's referenced in https://github.com/nerves-project/nerves_bootstrap/blob/master/lib/nerves_bootstrap/aliases.ex#L6 so it's mandatory. 

At the moment, this seems unavoidable, but it would be nice to get back to where it wasn't required.

**Comment by mobileoverlord** [comment]
This isn't entirely true. The current stipulation is that nerves_bootstrap has been started. for example, this mix file works perfectly fine

```
defmodule Test.MixProject do
  use Mix.Project
  
  Application.start(:nerves_bootstrap)

  def project do
    [
      app: :test,
      version: "0.1.0",
      elixir: "~> 1.4",
      archives: [nerves_bootstrap: "~> 1.0-rc"],
      build_embedded: Mix.env == :prod,
      start_permanent: Mix.env == :prod,
      aliases: Nerves.Bootstrap.add_aliases([]),
      deps: deps()
    ]
  end

  # Run "mix help compile.app" to learn about applications.
  def application do
    [mod: {Test.Application, []}, extra_applications: [:logger]]
  end
  
  # Run "mix help deps" to learn about dependencies.
  defp deps do
    [
      {:nerves, "~> 1.0-rc", override: true, runtime: false},
      {:shoehorn, "~> 0.2"},
      {:nerves_runtime, "~> 0.4"},
      {:nerves_system_rpi0, ">= 0.0.0", runtime: false}
    ]
  end

end
```

While this seems short and sweet, it caused issues with being able to run mix tasks on the host that leveraged dependencies that had native compiled code, like `fs`. The user would also have trouble entering into an `iex` shell with the project loaded.

**Comment by fhunleth** [comment]
The issue that I'm trying to raise is that we shouldn't reference `MIX_TARGET` in `nerves_bootstrap` at all. Since all of its values are user-defined in the user's `mix.exs`, code here or in libraries can't make any assumptions about values. The best I think that can derive from it is whether someone is intending to compile a Nerves-based project. Even then, it feels dicey to use. For example, before this change most of my projects didn't require that I set `MIX_TARGET`, since there was only one target that made sense. Now I set `MIX_TARGET` in my `mix.exs`, but that's a hack.

I'm not in a rush to get this fixed since I'm aware of the challenges, but I would like an issue to track it and inform others why they still need to set `MIX_TARGET` even if they only build for one system.
